### DISCUSSION TITLE
Create flowchart for model possible configurations

### DISCUSSION THREAD

**Discussion by iantaylor-NOAA** [NOAA-FIMS/FIMS#710] [Developers]
Here's a rough beginning of a flowchart illustrating potential model configurations as discussed in today's seaside chat.
@kellijohnson-NOAA suggested that @Bai-Li-NOAA and I could flesh this out and save it as a vignette. Since github-flavored markdown allows mermaid charts, I'm experimenting with editing the chart here in a github discussion where we can just edit this original post and use preview to debug expansions of this chart.

Feel free to edit this post and/or comment on the idea in separate comments.

```mermaid
flowchart TD

%% Nodes
    E("User runs FIMSFrame()")
    A{"check for weight-at-age data"}
    B("error")
    C{"check for length comps"}
    D{"check for age-to-<br>length conversion"}
   
%% Edge connections between nodes
    E --> A
    A -- no --> B
    A -- yes --> C
    C -- yes --> D
    D -- no --> B
    D -- yes --> ?

```

**Comment by kellijohnson-NOAA** [comment]
Thanks @iantaylor-NOAA for this example. I also wanted to highlight that there is an R package called pkgdepR that can create a map of functions dependencies. This might be useful for people trying to navigate from a high-level wrapper function to knowing which functions they will need to specify on their own. I am not sure how this functionality from {[pkgdepR](https://pkgdepr.org/)} can be useful to FIMS as of right now but I am just noting it here so I do not have to go through the trouble of googling which package to use again 😄.
